lib/active_merchant/billing/gateways/mondido.rb

```diff
@@ -0,0 +1,658 @@
+
+        options[:authorize] = false
+        create_post_for_auth_or_purchase(money, payment, options)
+      end
+
+      def authorize(money, payment, options={})
+        # Validate the credit card and reserve the money for later collection
+
+        options[:authorize] = true
+        create_post_for_auth_or_purchase(money, payment, options)
+      end
+
+      def capture(money, authorization, options={})
+        # References a previous “Authorize” and requests that the money be drawn down.
+        # It’s good practice (required) in many juristictions not to take a payment from a
+        #   customer until the goods are shipped.
+
+        put = {
+          # amount decimal *required 
+          #   The amount to refund. Ex. 5.00
+          :amount => get_amount(money, options)
+        }
+
+        add_support_params(put, options)
+        commit(:put, "transactions/#{authorization}/capture", put)
+      end
+
+      def refund(money, authorization, options={})
+        # Refund money to a card.
+        # This may need to be specifically enabled on your account and may not be supported by all gateways
+
+        requires!(options, :reason)
+        post = {
+          # transaction_id  int *required
+          #   ID for the transaction to refund
+          :transaction_id => authorization.to_i,
+
+          # amount decimal *required 
+          #   The amount to refund. Ex. 5.00
+          :amount => get_amount(money, options),
+
+          # reason string *required
+          #   The reason for the refund. Ex. "Cancelled order"
+          :reason => options[:reason]
+        }
+
+        add_support_params(post, options)
+        commit(:post, 'refunds', post)
+      end
+
+      def void(authorization, options={})
+        # Entirely void a transaction.
+        # Any amount into a refund will cancel the whole reservation
+
+        refund(100, authorization, options)
+      end
+
+      def verify(credit_card, options={})
+        # Test a payment authorizing a value of 1.00
+        # Then void the transaction and refund the value
+        options[:reason] ||= "Active Merchant Verify"
+
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      def store(payment, options = {})
+        post = {
+          # currency  string* required
+          :currency => options[:currency] || self.default_currency,
+
+          # test bool
+          #   Must be true if you are using a test card number.
+          :test => test?
+        }
+
+        # customer_ref  string
+        #   Merchant specific customer ID.
+        #   If this customer exists the card will be added to that customer.
+        #   If it doesn't exists a customer will be created.
+        post.merge!( :customer_ref => options[:customer_ref].to_s ) if options[:customer_ref]
```

Comment:
Whoops, I mean just `:customer`, as that's the standard AM option for specifying a customer profile identifier.